lib/function_object/arguments_builder.rb

```diff
@@ -12,8 +12,9 @@ def default?
     end
   end
 
-  def initialize
+  def initialize(args)
     @descriptors = []
+    arguments(args)
```

Comment:
I don't like this idea because it ties `ArgumentBuilder` to the syntax details of of `args` DSL.
Maybe `FunctionObject.args` became simpler but now `ArgumentsBuilder` became more complex. Complexity isn't gone, it just migrated to `ArgumentsBuilder` :) But this complexity is reasoned and this complexity should belong to DSL, not to `ArgumentsBuilder` because list of symbols passed in arguments is the concern of `FunctionObject.args`, and not the concern of `ArgumentsBuilder`.